### DISCUSSION TITLE
Magma Project Permissions: Changelist

### DISCUSSION THREAD

**Discussion by hcgatewood** [magma/magma#5866] [General]
# Magma Project Permissions: Changelist

This Discussion is a companion to https://github.com/magma/magma/discussions/5784.

See below for permissions changelist that will go into effect based on the proposed changes. Date-of-change target is one week from when this Discussion was posted (so targeting Wed April 7).

Tables generated by https://csvtomd.com/.

## GitHub users

| GitHub username     | Common name          | Org (previous) | Org (new) | Repo (previous) | Repo (new) |
| ------------------- | -------------------- | -------------- | --------- | --------------- | ---------- |
| @arunuke            | Arun Thulasi         | Member         | Member    | Maintain        | Write      |
| @bbarritt            | Brian Barritt        | Owner          | Owner     | Write           | Admin      |
| @bigfootjon         | Jon Janzen           | Owner          | Member    | Write           | Write      |
| @ekowtaylor         | Ekow Taylor          | Owner          | Owner     | Write           | Admin      |
| @electronjoe        | Scott Moeller        | Member         | Member    | Maintain        | Admin      |
| @hcgatewood         | Hunter Gatewood      | Member         | Member    | Maintain        | Admin      |
| @jbryce             | Jonathan Bryce       | Owner          | Member    | Write           | Write      |
| @jeremy-donson      | Jeremy Donson        | Member         | Member    | Maintain        | Write      |
| @mcallahan          | Michael Callahan     | Member         | Member    | Admin           | Write      |
| @mpgermano          | Michael Germano      | Member         | Member    | Admin           | Write      |
| @PARitter           | Phil Ritter          | Owner          | Owner     | Write           | Admin      |
| @pshelar            | Pravin Shelar        | Member         | Member    | Maintain        | Admin      |
| @ranj8work          |                      | Member         | Member    | Maintain        | Write      |
| @rdefosse           | Raphael Defosseux    | Member         | Member    | Maintain        | Admin      |
| @sanazbazargan      | Sanaz Bazargan       | Owner          | Member    | Write           | Write      |
| @shaddi             | Shaddi Hasan         | Owner          | Member    | Write           | Write      |
| @shahrahman-fb      | Shah Rahman          | Owner          | Member    | Write           | Write      |
| @ssanadhya          | Shruti Sanadhya      | Owner          | Member    | Admin           | Write      |
| @syotfs             | Michael Cheng        | Owner          | Member    | Write           | Write      |
| @talkhasib          | Tariq Al-Khasib      | Owner          | Member    | Write           | Write      |
| @tmdzk              | Timothee Dzik        | Owner          | Member    | Admin           | Admin      |
| @ttx                | Thierry Carraz       | Owner          | Member    | Write           | Write      |
| @tykeal             | Andrew Grimberg      | Owner          | Member    | Write           | Write      |
| @ulaskozat          | Ulas Kozat           | Owner          | Member    | Admin           | Admin      |

## GitHub bots

Currently tracking down owners. Ideally `Member` and `Write` would be sufficient, but tbd.

Notes:

- [can't change] magmabot needs webhook management permissions, which requires `Admin`
- facebook-github-* can be reduced to Member+Read
- magma-docusaurus-bot is unclear, tbd

| GitHub name                    | Common name              | Org (previous)       | Repo (previous) |
| ------------------------------ | ------------------------ | -------------------- | --------------- |
| facebook-github-bot            | Facebook Community Bot   | Owner                | Write           |
| facebook-github-permission-bot | Facebook Permission Bot  | Owner                | Write           |
| magma-docusaurus-bot           | Facebook Docussaurus Bot | Outside Collaborator | Write           |
| magmabot                       | Magmabot                 | Member               | Admin           |

**Comment by electronjoe** [comment]
Thank you for managing this process @hcgatewood, and thanks @PARitter for putting in the effort to get this kickstarted.

**Comment by bigfootjon** [comment]
We can remove `facebook-github-bot` and `facebook-github-permissions-bot` completely. Since this org is no longer managed by Facebook, we don't need our helper bots to be owners (or members) of the org.

CC @zpao: can you confirm the above?

Also, feel free to remove my account (@bigfootjon), as I don't need access anymore either :)

**Reply by bbarritt** [reply]
We want to retain the activity feed / metrics for Facebook employee's Github contributions.  Would removing the bots negatively impact those systems, @bigfootjon ?

**Reply by bigfootjon** [reply]
Ah right. In that case, @PARitter we will need the bots restored as members. (I can remain removed)

Additionally, please keep the `Facebook GitHub Tools` GitHub App installed on this org, as our internal systems make certain classes of API calls through this App.

**Comment by PARitter** [comment]
Done, per your request.

Phil Ritter
***@***.***

________________________________

**Comment by PARitter** [comment]
What is their appropriate access level?

________________________________

**Reply by bigfootjon** [reply]
Member with read-only access should be fine

**Comment by PARitter** [comment]
@facebook-github-bot and @facebook-github-permission-bot have both been re-invited to the org but someone will need to go to the github accounts for them and accept their invites.

Phil

________________________________

**Reply by bigfootjon** [reply]
I'll login and accept shortly

**Comment by PARitter** [comment]
I don’t see that the app “Facebook GitHub Tools” has ever had access.  There is an app “Facebook Intern Tools” installed.  Same thing?  Or is there something we need to correct?

Phil

________________________________

**Reply by bigfootjon** [reply]
I think that should be fine. If not, we can follow up
